Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java

```diff
@@ -412,7 +410,46 @@ public String getHashToken(String trustStorePasskey) throws CertificateManagerEx
         } catch (Exception e) {
             throw new CertificateManagerException("Error while fetching certificates-chain.", e);
         }
-        return soapResponseToDTO(response.getCertList());
+        return sortChainCertificate(response.getCertList(), alias);
+    }
+
+    private List<CertificateDTO> sortChainCertificate(List<ListCertificateType> soapList, String alias) throws CertificateManagerException{
+        List<CertificateDTO> tempList = soapResponseToDTO(soapList);
+        List<CertificateDTO> retList = new LinkedList<>();
+
+        CertificateDTO child = addCertByAlias(retList, tempList, alias);
+
+        while(null != child){
+            child = addCertByChild(retList, tempList, child);
+        }
+        return retList;
+
+    }
+
+    private CertificateDTO addCertByAlias(List<CertificateDTO> toList, List<CertificateDTO> fromList, String alias){
+        for(CertificateDTO cert : fromList){
+            if (StringUtils.isNotEmpty(cert.getAlias()) && StringUtils.isNotEmpty(alias)
```

Comment:
Optional: We can group all the comparation into 1 method equalsIgnoreCase(string1,string2).